pkg/pillar/cmd/client/client.go

```diff
@@ -65,54 +65,56 @@ var Version = "No version specified"
 //
 
 type clientContext struct {
+	agentBaseContext       agentbase.Context
 	subDeviceNetworkStatus pubsub.Subscription
 	deviceNetworkStatus    *types.DeviceNetworkStatus
 	usableAddressCount     int
 	subGlobalConfig        pubsub.Subscription
 	globalConfig           *types.ConfigItemValueMap
 	zedcloudCtx            *zedcloud.ZedCloudContext
 	getCertsTimer          *time.Timer
+	//CLI Args
+	noPidFlag  bool
+	maxRetries int
 }
 
 var (
-	debug             = false
-	debugOverride     bool // From command line arg
 	serverNameAndPort string
 	serverName        string
 	onboardTLSConfig  *tls.Config
 	devtlsConfig      *tls.Config
```

Comment:
I think these can be moved into clientCtxPtr

clientCtxPtr should be the only global variable in each agent..